Source/WebKit/Scripts/webkit/tests/GeneratedWebKitSecureCoding.cpp

```diff
@@ -0,0 +1,119 @@
+    : m_propertyList([object _webKitPropertyListData])
+{
+}
+
+bool CoreIPCNSSomeFoundationType::isValidDictionary(CoreIPCDictionary& dictionary)
+{
+    if (!dictionary.keyHasValueOfType("StringKey"_s, IPC::NSType::String))
+        return false;
+
+    if (!dictionary.keyHasValueOfType("NumberKey"_s, IPC::NSType::Number))
+        return false;
+
+    if (!dictionary.keyIsMissingOrHasValueOfType("OptionalNumberKey"_s, IPC::NSType::Number))
+        return false;
+
+    if (!dictionary.keyHasValueOfType("ArrayKey"_s, IPC::NSType::Array))
+        return false;
+
+    if (!dictionary.keyIsMissingOrHasValueOfType("OptionalArrayKey"_s, IPC::NSType::Array))
+        return false;
+
+    if (!dictionary.keyHasValueOfType("DictionaryKey"_s, IPC::NSType::Dictionary))
+        return false;
+
+    if (!dictionary.keyIsMissingOrHasValueOfType("OptionalDictionaryKey"_s, IPC::NSType::Dictionary))
+        return false;
+
+    return true;
+}
+
+RetainPtr<id> CoreIPCNSSomeFoundationType::toID() const
+{
+    RELEASE_ASSERT([NSSomeFoundationType instancesRespondToSelector:@selector(_initWithWebKitPropertyListData:)]);
+    return adoptNS([[NSSomeFoundationType alloc] _initWithWebKitPropertyListData:m_propertyList.toID().get()]);
+}
+
+#if ENABLE(DATA_DETECTION)
+CoreIPCDDScannerResult::CoreIPCDDScannerResult(DDScannerResult *object)
+    : m_propertyList([object _webKitPropertyListData])
+{
+}
+
+bool CoreIPCDDScannerResult::isValidDictionary(CoreIPCDictionary& dictionary)
+{
+    if (!dictionary.keyHasValueOfType("StringKey"_s, IPC::NSType::String))
+        return false;
+
+    if (!dictionary.keyHasValueOfType("NumberKey"_s, IPC::NSType::Number))
+        return false;
+
+    if (!dictionary.keyIsMissingOrHasValueOfType("OptionalNumberKey"_s, IPC::NSType::Number))
+        return false;
+
+    if (!dictionary.keyHasValueOfType("ArrayKey"_s, IPC::NSType::Array))
+        return false;
+    if (!dictionary.collectionValuesAreOfType("ArrayKey"_s, IPC::NSType::DDScannerResult))
+        return false;
+
+    if (!dictionary.keyIsMissingOrHasValueOfType("OptionalArrayKey"_s, IPC::NSType::Array))
+        return false;
+    if (!dictionary.collectionValuesAreOfType("OptionalArrayKey"_s, IPC::NSType::DDScannerResult))
+        return false;
+
+    if (!dictionary.keyHasValueOfType("DictionaryKey"_s, IPC::NSType::Dictionary))
+        return false;
+    if (!dictionary.collectionValuesAreOfType("DictionaryKey"_s, IPC::NSType::String, IPC::NSType::Number))
+        return false;
+
+    if (!dictionary.keyIsMissingOrHasValueOfType("OptionalDictionaryKey"_s, IPC::NSType::Dictionary))
+        return false;
+    if (!dictionary.collectionValuesAreOfType("OptionalDictionaryKey"_s, IPC::NSType::String, IPC::NSType::DDScannerResult))
+        return false;
+
+    return true;
+}
+
+RetainPtr<id> CoreIPCDDScannerResult::toID() const
+{
+    RELEASE_ASSERT([PAL::getDDScannerResultClass() instancesRespondToSelector:@selector(_initWithWebKitPropertyListData:)]);
```

Comment:
I feel like we're going to need a call to PAL::isDataDetectorsCoreFrameworkAvailable() or we'll break the base system.

Reply:
I had thought of this and dug in a little.

tldr: Since the base system already never does any data detector operations, we'll already never be encoding or decoding these.